### PR TITLE
[Merged by Bors] - Consolidate top level config

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- reworked `clusterConfig` and `authentication` according to https://github.com/stackabletech/issues/issues/293

fixes https://github.com/stackabletech/superset-operator/issues/339

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by maltesander** [comment]
test https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/superset-operator-it-custom/43/

File: tests/templates/kuttl/ldap/5-install-superset.yaml.j2 [hunk 1]
```diff
@@ -31,24 +31,25 @@ spec:
...
-    authenticationClass: {% if test_scenario['values']['ldap-authentication'] == 'no-tls' -%}
-        superset-with-ldap-no-tls-ldap
-      {%- elif test_scenario['values']['ldap-authentication'] == 'insecure-tls' -%}
-        superset-with-ldap-insecure-tls-ldap
-      {%- elif test_scenario['values']['ldap-authentication'] == 'server-verification-tls' -%}
-        superset-with-ldap-server-veri-tls-ldap
-      {%- endif %}
-
-    userRegistrationRole: Admin
+        replicas: 1
```
Comment (razvan) [inline tests/templates/kuttl/ldap/5-install-superset.yaml.j2:55]: we could just remove it completely

**Review by razvan — APPROVED** [review]
lgtm!

**Comment by razvan** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/superset-operator-it-custom/44/

File: tests/templates/kuttl/smoke/30-install-superset.yaml.j2 [hunk 2]
```diff
@@ -27,11 +27,12 @@ spec:
   image:
     productVersion: "{{ test_scenario['values']['superset'].split('-stackable')[0] }}"
     stackableVersion: "{{ test_scenario['values']['superset'].split('-stackable')[1] }}"
+  clusterConfig:
+    credentialsSecret: superset-credentials
```
[the remark this answers is not in the document]
Comment (maltesander) [reply tests/templates/kuttl/smoke/30-install-superset.yaml.j2:31]: Decided to merge as is for now and work on the static auth class later.

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**